vault/resource_pki_secret_backend_key.go

```diff
@@ -0,0 +1,265 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package vault
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"regexp"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/hashicorp/terraform-provider-vault/internal/consts"
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
+)
+
+var (
+	kmsType = "kms"
+
+	pkiSecretMountFromPathRegex = regexp.MustCompile("^(.+)/key/.+$")
+	pkiSecretKeyIDFromPathRegex = regexp.MustCompile("^.+/key/(.+)$")
+)
+
+func pkiSecretBackendKeyResource() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: MountCreateContextWrapper(pkiSecretBackendKeyCreate, provider.VaultVersion111),
+		UpdateContext: pkiSecretBackendKeyUpdate,
+		DeleteContext: pkiSecretBackendKeyDelete,
+		ReadContext:   ReadContextWrapper(ReadContextWrapper(pkiSecretBackendKeyRead)),
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			consts.FieldMount: {
```

Comment:
@raymonstah and I had a discussion on this and it would be great to get your thoughts @vinay-gopalan. Do you recommend that all non-mounting resources like this one use `FieldMount` and all mounting resources (something like our new `ldap_secret_backend`) should use `FieldPath`? It appears as thought that is the pattern you are following here.

Reply:
@fairclothjm good eye and thanks for calling that out! That is exactly right; this also ties in with a similar discussion I had with @raymonstah some time ago around deprecating uses of `FieldBackend`, and using `FieldMount` and `FieldPath` in the ways you described above!

Reply:
Nice, thanks for confirming! I will make sure to try to enforce this pattern going forward.